### DISCUSSION TITLE
New ideas for DSA

### DISCUSSION THREAD

**Discussion by TawfikYasser** [MakeContributions/DSA#222] [Ideas]
Hi

Welcome all to this great community

As you know we are all now working on re document the repository according to algorithms [here](https://github.com/MakeContributions/DSA/discussions/206)

And I've new ideas

* Include README file to define the **Author** of the code, the **Date** , to document the process of contribution. [Meta Data]
* Create new part in **DSA** to let problem solvers to add their programs to GitHub.
* Make a **Plan** and assign the work for the reviewers to review the code and make it more readable and more correct and assign any needed work to the code's author.
* Set strict rules on contributing to avoid a lot of reviewing process and make a PR template and Issue template, Naming and more, I know that a lot of these rules in contributing rules but a lot of people don't follow them.

My vision to this repo to be a grate place and reference for new developers to learn and share code and help others.

This discussion can be opened for a long time to discuss and add more ideas and improve our great repo.

**Comment by ming-tsai** [comment]
We are just starting migrating, after migrating we will starting documentation

**Comment by ming-tsai** [comment]
1. I don't know why we should have meta data, because we have controller version
2. I don't understand, sorry could you make example?
3. We have reviewers list
4. I'm thinking on that, due to many reviewers on one pull request it will get confused

**Reply by ming-tsai** [reply]
@TawfikYasser

**Reply by TawfikYasser** [reply]
Hi @ming-tsai  Sorry for late

1. Okay you are right according controller version
2. I mean that we create a folder for example for people to add the solutions for problems they solve on hackerrank for example.
3. Okay, Yes
4. Her yes I really suggest to make rules for reviewers to make it more managed.

**Reply by ming-tsai** [reply]
I guess you are talking about ideahub, I have json for projects and ideas, so that the user could use it as put the implementation
